Wikipedia/Code/AccountCreationViewController.swift

```diff
@@ -0,0 +1,285 @@
+
+import UIKit
+
+class AccountCreationViewController: UIViewController, CaptchaViewControllerRefresh, UITextFieldDelegate, UIScrollViewDelegate {
+    @IBOutlet var usernameField: UITextField!
+    @IBOutlet var passwordField: UITextField!
+    @IBOutlet var passwordRepeatField: UITextField!
+    @IBOutlet var emailField: UITextField!
+    @IBOutlet var captchaContainer: UIView!
+    @IBOutlet var scrollView: UIScrollView!
+    @IBOutlet var loginButton: UILabel!
+    @IBOutlet var titleLabel: UILabel!
+    @IBOutlet var usernameUnderlineHeight: NSLayoutConstraint!
+    @IBOutlet var passwordUnderlineHeight: NSLayoutConstraint!
+    @IBOutlet var passwordConfirmUnderlineHeight: NSLayoutConstraint!
+    @IBOutlet var emailUnderlineHeight: NSLayoutConstraint!
+    @IBOutlet var spaceBeneathCaptchaContainer: NSLayoutConstraint!
+    @IBOutlet var createAccountContainerView: UIView!
+
+    fileprivate var captchaId: NSString? = ""
+    fileprivate var rightButton: UIBarButtonItem?
+    public var funnel: CreateAccountFunnel?
+    fileprivate var captchaViewController: CaptchaViewController?
+    
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        // Disable horizontal scrolling.
+        scrollView.contentOffset = CGPoint(x: 0.0, y: scrollView.contentOffset.y)
+    }
+    
+    fileprivate func adjustScrollLimitForCaptchaVisiblity() {
+        // Reminder: spaceBeneathCaptchaContainer constraint is space *below* captcha container -
+        // that's why below for the show case we don't have to "convertPoint".
+        spaceBeneathCaptchaContainer.constant = (showCaptchaContainer)
+            ? (view.frame.size.height - (captchaContainer.frame.size.height / 2))
+            : (view.frame.size.height - loginButton.convert(CGPoint.zero, to:scrollView).y)
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        // Ensure adjustScrollLimitForCaptchaVisiblity gets called again after rotating.
```

Comment:
Won't this be called before rotating? (viewWILLTransistion) Maybe thats ok, just noticed.

Also if you want the change to be animated you can call it within the coordinater's animation block.